discord/role.py

```diff
@@ -40,50 +41,24 @@ def __repr__(self):
 
 
 class Role(DiscordObject):
-    __slots__ = (
-        "_guild",
-        "_cache",
-        "_id",
-        "_name",
-        "_color",
-        "_hoist",
-        "_position",
-        "_permissions",
-        "_managed",
-        "_mentionable",
-        "_tags",
-    )
-
-    _guild: Guild
-    _cache: RoleCache
-    _id: Snowflake
-    _name: str
-    _color: Color
-    _hoist: bool
-    _position: int
-    _permissions: str
-    _managed: bool
-    _mentionable: bool
-    _tags: Optional[RoleTags]
-
-    def __init__(self, guild: Guild, cache: RoleCache, payload: RolePayload):
-        self._guild = guild
-        self._cache = cache
-        self._id = payload["id"]
-        self._name = payload["name"]
-        self._color = payload["color"]
-        self._hoist = payload["hoist"]
-        self._position = payload["position"]
-        self._permissions = payload["permissions"]
-        self._managed = payload["managed"]
-        self._mentionable = payload["mentionable"]
-        self._tags = RoleTags(payload["tags"])
+    id: Snowflake
```

Comment:
Imo having these properties be private is better since we want them to be read-only